javascript/javascript3/week2/currencyCalculator.js

```diff
@@ -0,0 +1,64 @@
+const currencies = [];
+const ratesStorage = {};
+
+const fromSelect = document.getElementById("from-currency");
+const toSelect = document.getElementById("to-currency");
+
+const inputTag = document.getElementById("from-amount");
+const toConverted = document.getElementById("to-amount");
+
+async function fetchCurrencyRates(base = "EUR") {
+    try {
+        const res = await fetch(`https://open.er-api.com/v6/latest/${base}`);
+        const data = await res.json();
+
+        currencies.push(...Object.keys(data.rates));
+
+        return data;
+    } catch (err) {
+        console.error(`Something went wrong: ${err}`);
+    } 
+}
+
+async function generateOptions() {
+    await fetchCurrencyRates();
+    const htmlOptions = currencies
+    .map((option) => {
+        return `<option value="${option}">${option}</option>`;
```

Comment:
You will really enjoy the React module, but for now, modifying the innerHTML with code from strings isn't the best way to do this. Instead, use the [createElement DOM method](https://developer.mozilla.org/en-US/docs/Web/API/Document/createElement) and modify its value property. Then use the [appendChild DOM method](https://developer.mozilla.org/en-US/docs/Web/API/Node/appendChild) to add this element to the HTML.